### PR TITLE
Add a coloring how-to

### PR DESCRIPTION
**Pull request by caleb-johnson**
This PR adds a how-to showing how coloring can be used to improve the depth of circuits by more efficiently placing gates.

### CODE REVIEW

**Comment by caleb-johnson** [comment]
@mrossinek I can probably get graphviz installed in the CI envs, but I'm a bit confused why OBP can use graphviz_draw in the notebooks using the same type of dependency specification of rustworkx[graphviz] that I do here.

**Review by mrossinek — COMMENTED** [review]
Overall I like this, thanks for putting it together!

Just some nitpicky ideas :+1:

File: docs/how_tos/color_device_edges_to_improve_depth.ipynb
[the remark this answers is not in the document]
Comment (caleb-johnson) [reply docs/how_tos/color_device_edges_to_improve_depth.ipynb:1]: It should be trivial to add the barriers to the second circuit. I could use `slicing` and `combine_slices` to get barriers in the first one :)

Comment (mrossinek) [inline docs/how_tos/color_device_edges_to_improve_depth.ipynb:1]: Okay, final idea: now that you use `slice_by_depth` for the first circuit but insert the barriers manually in the second case, would it not be cool to also use the `slice_by_depth` method there? That shows that we are indeed not "cheating" and that the same automated slicing method arrives at vastly different number of slices.

**Review by mrossinek — APPROVED** [review]

**Pull request merged**